docs: add docs for vulnerability related commands (#3362)
canonical/ubuntu-pro-client: docs/howtoguides/how_to_update_vulnerability_data.rst

> How to update vulnerability data on the machine?

Reviewer: If you do decide to roll these pages together, I think this should be the first one. In that case, I suggest adding:
* a general introduction paragraph that shows the ``pro vulnerability list`` and ``pro vulnerability`` show commands
* a brief outline of what situation each is useful for (primarily geared towards CVEs since that's the main thing people will be interested in),
* and then links to each of the explanation pages at the top of the page.

People will be primarily looking for a "how to use the pro vulnerability" command page, so that should be the first part of the content we show, with these related topics underneath.

Author: I don't think this is a bad idea, but let's discuss it better, specially for the sections we have in the `howtoguides`

Reviewer: Discussed side-channel, leaving as is for now

************************************************

When running the ``pro vulnerability list`` you might see an alert like this:

.. code-block:: text

    The vulnerabilities data used in the system is outdated by NUM days/hours
    To update the data please run with --update:

        pro vulnerability list --update

> This same behavior can happen when running ``pro vulnerability show``. This indicates
> that our data source for the vulnerability commands is outdated in the machine.

Reviewer: ```suggestion
This behaviour can happen when running ``pro vulnerability show``, and it indicates
that our data source for the vulnerability commands is outdated in the machine.
```

To update it, you can follow the instructions mentioned in the warning or you

> can run the following command as well:

Reviewer: ```suggestion
can run the following command:
```

.. code-block:: bash

    pro vulnerability update

> The main purpose of this command is to just allow you to update that datasource.

Reviewer: ```suggestion
The main purpose of this command is to only update that datasource.
```
